controllers/azureidentity_controller_test.go

```diff
@@ -0,0 +1,132 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	aadpodv1 "github.com/Azure/aad-pod-identity/pkg/apis/aadpodidentity/v1"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-azure/util/system"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	capifeature "sigs.k8s.io/cluster-api/feature"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestAzureIdentityControllerReconcileAzureCluster(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	req := ctrl.Request{NamespacedName: types.NamespacedName{Name: "fake-cluster", Namespace: "fake-ns"}}
+	scheme := runtime.NewScheme()
+	err := infrav1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+	err = aadpodv1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	azureCluster := &infrav1.AzureCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-cluster",
+			Namespace: "fake-ns",
+		},
+	}
+
+	bindings := &aadpodv1.AzureIdentityBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-binding",
+			Namespace: system.GetManagerNamespace(),
+			Labels: map[string]string{
+				clusterv1.ClusterNameLabel:    "another-fake-cluster",
+				infrav1.ClusterLabelNamespace: "fake-ns",
+			},
+		},
+		Spec: aadpodv1.AzureIdentityBindingSpec{
+			AzureIdentity: "managedIdentity",
+		},
+	}
+
+	azIdentity := &aadpodv1.AzureIdentity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "managedIdentity",
+			Namespace: system.GetManagerNamespace(),
+		},
+	}
+
+	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(azureCluster, bindings, azIdentity).Build()
+	aiRec := &AzureIdentityReconciler{
+		Client:           client,
+		Recorder:         record.NewFakeRecorder(42),
+		ReconcileTimeout: 20 * time.Second,
+		WatchFilterValue: "fake",
+	}
+	_, err = aiRec.Reconcile(ctx, req)
+	g.Expect(err).ToNot(HaveOccurred())
+}
+
+func TestAzureIdentityControllerReconcileAzureManagedControlPlane(t *testing.T) {
+	g := NewWithT(t)
+	defer featuregatetesting.SetFeatureGateDuringTest(t, capifeature.Gates, capifeature.MachinePool, true)()
+	ctx := context.Background()
+	req := ctrl.Request{NamespacedName: types.NamespacedName{Name: "fake-controlplane", Namespace: "fake-ns"}}
+	scheme := runtime.NewScheme()
```

Comment:
Same comment as above ^

Reply:
Pushed on latest update and added `aadpodv1` to the newScheme() function.  Resolving.